packages/xrpl/src/Wallet/signer.ts

```diff
@@ -94,8 +94,12 @@ function verifySignature(tx: Transaction | string): boolean {
   const decodedTx: Transaction = getDecodedTransaction(tx)
   return verify(
     encodeForSigning(decodedTx),
-    decodedTx.TxnSignature,
-    decodedTx.SigningPubKey,
+    // Need a SignedTransaction class where TxnSignature is not optional
+    // Verify signature used to be untyped
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- See above
+    decodedTx.TxnSignature!,
```

Comment:
TODO

Reply:
I combined most aspects of `Wallet.verifyTransaction` and `verifySignature` so they are treated similarly and added a test.